### DISCUSSION TITLE
Get the types of extendTailwindMerge

### DISCUSSION THREAD

**Discussion by stijns96** [dcastil/tailwind-merge#501] [Help]
I'm creating a module in Nuxt 3 where I use the app.config.ts to get certain configuration from the user. In this file I want the user to extend the tailwind merge config.

It looks like this

**app.config.ts**
export default defineAppConfig({
  testModule: {
    tailwindMerge: {
      extend: {
        theme: {
          spacing: [
            'sm',
            'base',
            'lg'
          ]
        }
      }
    }
  }
})

Of course I want this to be typed to make it easier, but I can't find out how to make this work, since certain types are not exported.

On [this line](https://github.com/dcastil/tailwind-merge/blob/v2.5.5/src/lib/types.ts#L112) I can find the actual interface that I need (if I'm right).

How can I make sure to use this in my module?

**Comment by stijns96** [comment]
Dived into it a little deeper and found a workaround.

```js
import { extendTailwindMerge } from 'tailwind-merge'
tailwindMerge?: Parameters<typeof extendTailwindMerge>[0]
```

**Comment by dcastil** [comment]
Hey @stijns96! 👋

Indeed makes sense to export it! I added the issue https://github.com/dcastil/tailwind-merge/issues/504 to track that and will implement it soon. But your approach with `Parameters<typeof extendTailwindMerge>[0]` should work in the meantime.

**Reply by stijns96** [reply]
Great! I do have one issue with it. It doesn't know about the values deeper inside.

**This is what is expected, but what is missing with the mentioned approach.**
<img width="709" alt="image" src="https://github.com/user-attachments/assets/0416525e-87a0-48f5-bded-858406547a79" />
